schema.go

```diff
@@ -579,6 +652,22 @@ func (s *RecordSchema) FingerprintUsing(typ FingerprintType) ([]byte, error) {
 	return s.fingerprinter.FingerprintUsing(typ, s)
 }
 
+// CacheFingerprint returns a special fingerprint of the schema for caching purposes.
+func (s *RecordSchema) CacheFingerprint() [32]byte {
+	data := make([]any, 0)
+	for _, field := range s.fields {
+		if field.Default() != nil {
```

Comment:
Is this correct? `nil` is a valid default. I would have thought it should check `HasDefault` here, and then deal with `nil`?

Reply:
You are right! `nil` is a valid default that should be part of the cache's fingerprint.